src/ert/_c_wrappers/enkf/res_config.py

```diff
@@ -405,11 +420,88 @@ def _alloc_from_dict(self, config_dict):
 
         self.model_config = ModelConfig(
             data_root=self.config_path,
-            joblist=self.site_config.job_list,
+            joblist=self.job_list,
             refcase=self.ensemble_config.refcase,
             config_dict=config_dict,
         )
 
+    def _read_env_vars(self, user_config_dict, site_config_dict):
+        env_vars = EnvironmentVarlist()
+
+        def add_from_dict(config_dict):
+            environment_vars = config_dict.get(ConfigKeys.SETENV, [])
+
+            for item in iter(environment_vars):
+                if isinstance(item, Mapping):
+                    key = item["NAME"]
+                    value = item["VALUE"]
+                else:
+                    key, value = item
```

Comment:
One is for reading from the config dict and one from the config content

Reply:
by from config content you mean the dict generated by config content's `from_dict` method? i would expect that we make those equal, where we use them, but maybe that's a bit premature?

Reply:
I have no idea what we are aiming for there. Perhaps @eivindjahren wants to chime in

Reply:
Eivind says we can change the config dict generator to just create a two element list, instead of having a dictionary with NAME and VALUE keys. then it will mirror what `_config_content_as_dict` produces

Reply:
Generally think there should be one dict to rule them all, no `as_dict`, `config_dict`, `from_dict`, `everest_dict` doing different things. However, we can move towards that without making all the changes at once.

Reply:
yeah so I'll leave it at your discretion, Morten, if you wanna change the structure of the config dict generator, or just leave it as is : )